go/vt/vtgateproxy/discovery.go

```diff
@@ -323,7 +323,9 @@ func (b *JSONGateResolverBuilder) parse() (bool, error) {
 	for poolType := range targets {
 		b.sorter.shuffleSort(targets[poolType], b.affinityField, b.affinityValue)
 		if len(targets[poolType]) > *numConnections {
```

Comment:
nit: preexisting code and no change in behavior, but this should be `if len(targets[poolType]) > b.numConnections {`